wai-extra/Network/Wai/Middleware/CombineHeaders.hs

```diff
@@ -0,0 +1,280 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RecordWildCards #-}
+{- |
+Sometimes incoming requests don't stick to the
+"no duplicate headers" invariant, for a number
+of possible reasons (e.g. proxy servers blindly
+adding headers), or your application (or other
+middleware) blindly adds headers.
+
+In those cases, you can use this 'Middleware'
+to make sure that headers that /can/ be combined
+/are/ combined. (e.g. applications might only
+check the first \"Accept\" header and fail, while
+there might be another one that would match)
+    -}
+module Network.Wai.Middleware.CombineHeaders
+    ( combineHeaders
+    , CombineSettings
+    , defaultCombineSettings
+    , HeaderMap
+    , HandleType (..)
```

Comment:
Sorry, I had intended for the smart contract comment to apply to this type too.

Reply:
Done :)